src/main/java/model/Winning.java

```diff
@@ -0,0 +1,58 @@
+package model;
+
+import java.util.Arrays;
+import java.util.Optional;
+
+public enum Winning {
+    FAIL(0, 0, 0, ""),
+    FIFTH(3, 5000, 0, "3개 일치 (5,000원)"),
+    FOURTH(4, 50000, 0, "4개 일치 (50,000원)"),
+    THIRD(5, 1500000, 0, "5개 일치 (1,500,000원)"),
+    SECOND(5, 30000000, 1, "5개 일치, 보너스 볼 일치 (30,000,000원)"),
+    FIRST(6, 2000000000, 0, "6개 일치 (2,000,000,000원)");
+
+    private final int count;
+    private final long prize;
+    private final int hasBonusNumber;
+    private final String description;
+
+    Winning(int count, long prize, int hasBonusNumber, String description) {
+        this.count = count;
+        this.prize = prize;
+        this.hasBonusNumber = hasBonusNumber;
+        this.description = description;
+    }
+
+    public static Winning matchWinning(int count, int hasBonusNumber) {
+        Winning result = Arrays.stream(Winning.values())
+                .filter(w -> w.count == count)
+                .findFirst()
+                .orElse(Winning.FAIL);
+
+        if (result == THIRD && hasBonusNumber == SECOND.hasBonusNumber) {
+            result = SECOND;
+        }
+
+        return result;
+    }
+
+    public static long calculatePrize(Winning winning, int count) {
+        final Optional<Winning> target = Arrays.stream(Winning.values()).filter(w -> w.equals(winning)).findFirst();
+
+        if (target.isPresent()) {
+            return target.get().prize * count;
+        }
+
+        return 0;
+    }
+
+    public static String outputDescription(Winning winning) {
+        final Optional<Winning> target = Arrays.stream(Winning.values()).filter(w -> w.equals(winning)).findFirst();
+
+        if (target.isPresent()) {
+            return target.get().description;
+        }
+
+        return null;
+    }
```

Comment:
여기도 리팩토링을 하는게 좋을 것 같네요
```suggestion
            return target.map(value -> value.description)
                .orElse(null);
```

그리고 현재 `null` 를 반환을 하는데 과연 `null` 를 반환하는게 구조적으로 좋을지 생각해보셨으면 좋겠습니다.
`null` 을 직접 다룬다면 `NPE` 가 터질 수도 있고, 옵셔널을 사용하는데 `null` 를 직접 다루는건 이상하다고 생각됩니다~
자바에 옵셔널을 왜 생겼는지 공부해보고, 해당 로직 또한 리팩토링 하는게 좋을 것 같아요